tools/CMakeLists.txt

```diff
@@ -182,11 +182,17 @@ foreach (deploytgt drconfig drrun drinject)
     target_link_libraries(${deploytgt} drinjectlib)
   endif ()
   if (LINUX) # No checks for MacOS
-    get_target_property(check_path ${deploytgt} LOCATION)
-    add_custom_command(TARGET ${deploytgt}
-      POST_BUILD
+    set(locvar_rel "")
+    if ("${CMAKE_VERSION}" VERSION_EQUAL "3.10" OR
+        "${CMAKE_VERSION}" VERSION_GREATER "3.10")
+      set(locvar_rel "../")
+    endif()
+    set(locvar_name ${deploytgt}loc)
+    file(GENERATE OUTPUT "${locvar_rel}${locvar_name}.cmake" CONTENT "set(${locvar_name} \
+      \"$<TARGET_FILE:${deploytgt}>\")\n")
+    add_custom_command(TARGET ${deploytgt} POST_BUILD
       COMMAND ${CMAKE_COMMAND}
-      ARGS -D lib=${check_path}
+      ARGS -D lib_fileloc=${locvar_name}
```

Comment:
The relative dirs used make me nervous...I guess I'm echoing the comment up above.  So you're relying on the file generation and the script to have the same working dir, the current binary dir, so if we call CMake_readelf.cmake from tools/ and from core/ the file generation and the script are both in that subdir I suppose.  But any WORKING_DIRECTORY param seems like it will break it.  Absolute seems more robust.  Just repeating my above comment I guess.

Reply:
Same question above. Do you mean CMP0070 related?